pkg/backup/compaction_job.go

```diff
@@ -523,17 +523,46 @@ func updateCompactionBackupDetails(
 	return compactedDetails, nil
 }
 
-// compactIntroducedSpans takes a compacted backup manifest and the full chain of backups it belongs
-// to and computes the introduced spans for the compacted backup.
-func compactIntroducedSpans(
-	ctx context.Context, manifest backuppb.BackupManifest, chain compactionChain,
-) (roachpb.Spans, error) {
-	if err := checkCoverage(ctx, manifest.Spans, chain.backupChain); err != nil {
+// createCompactionManifest creates a new manifest for a compaction job and its
+// compacted chain. The details should have its targets resolved.
+func (c compactionChain) createCompactionManifest(
+	ctx context.Context, details jobspb.BackupDetails,
+) (*backuppb.BackupManifest, error) {
+	// TODO (kev-cao): Will need to update the SSTSinkKeyWriter to support
+	// range keys.
+	lastBackup := c.lastBackup()
+	if len(lastBackup.Tenants) != 0 {
+		return nil, errors.New("backup compactions does not yet support range keys")
+	}
+	introducedSpans, err := c.compactIntroducedSpans(ctx)
+	if err != nil {
+		return nil, err
+	}
+	cManifest := lastBackup
+	cManifest.StartTime = c.start
+	cManifest.EndTime = c.end
+	cManifest.Descriptors = details.ResolvedTargets
+	cManifest.CompleteDbs = details.ResolvedCompleteDbs
+	cManifest.IntroducedSpans = introducedSpans
+	cManifest.Files = nil
+	cManifest.DescriptorChanges = nil // Revision history is not supported.
+	cManifest.EntryCounts = roachpb.RowCount{}
+	cManifest.Dir = cloudpb.ExternalStorage{}
+	cManifest.HasExternalManifestSSTs = false
+	cManifest.PartitionDescriptorFilenames = nil // Locality aware backups not supported.
+	return &cManifest, nil
```

Comment:
i think you forgot to add the `IsCompacted` field? Given that no unit test caught this, i think we need to add a test in which you restore from a chain that contains compactions, with the setting on, and validate that the num uris in the restore details is different than when you run the same restore with the setting flipped off.

Reply:
i also wonder if this bug has been mucking with the investigation :D

Reply:
Yup I'll write a unit test for it. And fortunately, it did not interfere with the investigation because I wrote this PR to then absolutely never use it in my investigation and was just deleting backups by hand.

Reply:
In the process of writing this unit test, actually ended up finding another bug so resolved it and added the unit test in another PR: #144808